code/src/cljs/sixsq/nuvla/ui/session/events.cljs

```diff
@@ -89,6 +88,31 @@
      :dispatch [::clear-loading]}))
 
 
+(reg-event-fx
+  ::code-validation-2fa-failed
+  (fn []
+    {:fx [[:dispatch [::set-error-message "Code validation failed!"]]]}))
+
+
+(reg-event-fx
+  ::validate-2fa-activation
+  (fn [{{:keys [::spec/callback-2fa]} :db} [_ token]]
+    (when-not (str/blank? token)
+      {:http-xhrio {:method          :put
+                    :uri             callback-2fa
```

Comment:
if the code is not inserted shortly, the validation will fail with 
```
Error
cannot re-execute callback
```

does the token timeout? If so, maybe make that information explicit.

Reply:
if this error is supposed to happen (or can not be immediately fixed) then add a link to the modal to "resend code"